### PR TITLE
Mobileapps 1078

### PR DESCRIPTION
**Pull request by ankitgoyalgloballogic**
1. Database Migration
2. Allow user to search within app extension

### CODE REVIEW

File: Share/ListComponent/SystemSearchViewController.swift [hunk 1]
```diff
@@ -81,9 +81,15 @@ class SystemSearchViewController: SystemThemableViewController {
     // MARK: - IBActions
 
     @objc func searchButtonTapped() {
-        AlfrescoLog.debug("search button action")
+        navigationItem.searchController = searchController
+        searchController?.searchBar.alpha = 0.0
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.01) { [weak self] in
```
Comment (aman-alfresco) [inline Share/ListComponent/SystemSearchViewController.swift:86]: Do we add some delay like 0.01?

**Review by aman-alfresco — APPROVED** [review]

File: ContentApp/PersistenceLayer/Services/DatabaseMigrationService.swift [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                    nodeToBeUpload.id = 0
+                }
+                databaseService?.store(entity: nodeToBeUpload)
+                self.remove(entity: node)
+            }
+        }
+        
+        // downloaded files
+        let downloadedNodes = self.queryAll(entity: ListNode.self) ?? []
+        if !downloadedNodes.isEmpty {
```
Comment (gl-lovekesh) [inline ContentApp/PersistenceLayer/Services/DatabaseMigrationService.swift:66]: make a common function pass the nodes only.
func migrateNodes(nodes : ListNodes){
for node in downloadedNodes {
                let nodeToBeStored = node
                if node.id != 0 {
                    nodeToBeStored.id = 0
                }
                databaseService?.store(entity: nodeToBeStored)
                self.remove(entity: node)
            }
        }
}

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]

File: ContentApp/PersistenceLayer/Services/DatabaseMigrationService.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PersistenceLayer/Services/DatabaseMigrationService.swift:65]: Done

File: Share/ListComponent/SystemSearchViewController.swift (hunk 1 above)
Comment (gl-lovekesh) [reply Share/ListComponent/SystemSearchViewController.swift:86]: that is required for screen animation to be completed .

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**